components/brave_wallet/brave_wallet_service.cc

```diff
@@ -166,8 +168,15 @@ void BraveWalletService::SaveToPrefs(PrefService* prefs,
 }
 
 void BraveWalletService::ResetCryptoWallets() {
+  Profile* profile = Profile::FromBrowserContext(context_);
+  auto* registry = extensions::ExtensionRegistry::Get(context_);
+
   extensions::ExtensionPrefs::Get(context_)->DeleteExtensionPrefs(
       ethereum_remote_client_extension_id);
+
+  auto* erc_extension = registry->enabled_extensions().GetByID(
+      ethereum_remote_client_extension_id);
+  extensions::DataDeleter::StartDeleting(profile, erc_extension);
```

Comment:
I noticed this warning comment in the declaration of `StartDeleting`:

https://github.com/chromium/chromium/blob/09ec5f7b5120f6fa982779df93bc759cff338170/chrome/browser/extensions/data_deleter.h#L18-L21
```c
  // Starts removing data. The extension should not be running when this is
  // called. Cookies are deleted on the current thread, local storage and
  // databases/settings are deleted asynchronously on the webkit and file
  // threads, respectively. This function must be called from the UI thread.
```

Two questions:
- Do you know if this service runs in the UI thread? I'm not 100% clear on that personally so this is more for my own education :)
- Should we disable the ERC extension before calling this and causing some kind of undefined behavior in the extension itself?

Here's what the chromium extension service is doing.. wdyt?

https://github.com/chromium/chromium/blob/09ec5f7b5120f6fa982779df93bc759cff338170/chrome/browser/extensions/extension_service.cc#L812-L824
```c++
  UnloadExtension(extension->id(), UnloadedExtensionReason::UNINSTALL);
  if (registry_->blocklisted_extensions().Contains(extension->id()))
    registry_->RemoveBlocklisted(extension->id());


  // Tell the backend to start deleting installed extensions on the file thread.
  if (!Manifest::IsUnpackedLocation(extension->location())) {
    if (!GetExtensionFileTaskRunner()->PostTask(
            FROM_HERE,
            base::BindOnce(&ExtensionService::UninstallExtensionOnFileThread,
                           extension->id(), profile_, install_directory_,
                           extension->path())))
      NOTREACHED();
  }
```

Reply:
just to clarify, it looks like the line we probably care about most is:

```c++
UnloadExtension(extension->id(), UnloadedExtensionReason::UNINSTALL)
```

Since we're not uninstalling the extension, we can use another one of the reasons defined in the enum. It looks `DISABLE` may be our best. So in that case my suggestion would be something like

```suggestion
  extensions::ExtensionService* extension_service =
    extensions::ExtensionSystem::Get(profile)->extension_service();

  extension_service->UnloadExtension(erc_extension, UnloadedExtensionReason::DISABLE);
  extensions::DataDeleter::StartDeleting(profile, erc_extension);
```